discord/message.go

```diff
@@ -10,35 +10,32 @@ import (
 
 // https://discord.com/developers/docs/resources/channel#message-object
 type Message struct {
-	// ID is the id of the message.
-	ID MessageID `json:"id"`
-	// ChannelID is the id of the channel the message was sent in.
-	ChannelID ChannelID `json:"channel_id"`
-	// GuildID is the id of the guild the message was sent in.
-	GuildID GuildID `json:"guild_id,omitempty"`
-
-	// Type is the type of message.
-	Type MessageType `json:"type"`
-
-	// Flags are the MessageFlags.
-	Flags MessageFlags `json:"flags"`
-
-	// TTS specifies whether the was a TTS message.
-	TTS bool `json:"tts"`
-	// Pinned specifies whether the message is pinned.
-	Pinned bool `json:"pinned"`
-
-	// MentionEveryone specifies whether the message mentions everyone.
-	MentionEveryone bool `json:"mention_everyone"`
-	// Mentions contains the users specifically mentioned in the message.
+	// EditedTimestamp specifies when this message was edited.
 	//
-	// The user objects in the mentions array will only have the partial
-	// member field present in MESSAGE_CREATE and MESSAGE_UPDATE events from
-	// text-based guild channels.
-	Mentions []GuildUser `json:"mentions"`
-	// MentionRoleIDs contains the ids of the roles specifically mentioned in
-	// the message.
-	MentionRoleIDs []RoleID `json:"mention_roles"`
+	// IsValid() will return false, if the messages hasn't been edited.
+	EditedTimestamp Timestamp `json:"edited_timestamp,omitempty"`
+	// Timestamp specifies when the message was sent
+	Timestamp Timestamp `json:"timestamp,omitempty"`
+	// Reference is the reference data sent with crossposted messages and
+	// inline replies.
+	Reference *MessageReference `json:"message_reference,omitempty"`
+	// Application is sent with Rich Presence-related chat embeds.
+	Application *MessageApplication `json:"application,omitempty"`
+	// Activity is sent with Rich Presence-related chat embeds.
+	Activity *MessageActivity `json:"activity,omitempty"`
+	// ReferencedMessage is the message that was replied to. If not present and
+	// the type is InlinedReplyMessage, the backend couldn't fetch the
+	// replied-to message. If null, the message was deleted. If present and
+	// non-null, it is a message object
+	ReferencedMessage *Message `json:"referenced_message,omitempty"`
```

Comment:
This is unrelated to this PR, but this comment should be changed to the below, since no distinction between omitted and nil is possible.

```suggestion
	// ReferencedMessage is the message that was replied to.
	ReferencedMessage *Message `json:"referenced_message,omitempty"`
```